src/playlist-maker.js

```diff
@@ -133,217 +122,336 @@ const factory = (player, initialList, initialIndex = 0) => {
+      return list[playlist.currentItem(0)];
+    }
 
-      // Make the change
-      if (nextIndex !== playlist.currentIndex_) {
-        return list[playlist.currentItem(nextIndex)];
-      }
-    },
+    playlist.currentIndex_ = -1;
+  };
+
+  /**
+   * Plays the last item in the playlist.
+   *
+   * @return {Object|undefined}
+   *         Returns undefined and has no side effects if the list is empty.
+   */
+  playlist.last = () => {
+    if (list.length) {
+      return list[playlist.currentItem(playlist.indexOfLast())];
+    }
+
+    playlist.currentIndex_ = -1;
+  };
+
+  /**
+   * Plays the next item in the playlist.
+   *
+   * @return {Object|undefined}
+   *         Returns undefined and has no side effects if on last item.
+   */
+  playlist.next = () => {
+    const index = playlist.indexOfNext();
+
+    if (index !== playlist.currentIndex_) {
+      return list[playlist.currentItem(index)];
+    }
+  };
+
+  /**
+   * Plays the previous item in the playlist.
+   *
+   * @return {Object|undefined}
+   *         Returns undefined and has no side effects if on first item.
+   */
+  playlist.previous = () => {
+    const index = playlist.indexOfPrevious();
+
+    if (index !== playlist.currentIndex_) {
+      return list[playlist.currentItem(index)];
+    }
+  };
+
+  /**
+   * Get/set auto-advance on the playlist.
+   *
+   * @param  {number} [delay]
+   *         The number of seconds to wait before each auto-advance.
+   *         Leave this argument off to get the current auto-advance delay.
+   *
+   * @return {number|null}
+   *         The current delay.
+   */
+  playlist.autoadvance = (delay) => {
+    if (delay !== undefined) {
+      autoadvance.setup(playlist.player_, delay);
+    }
+  };
+
+  /**
+   * Sets `repeat` option, which makes the "next" video of the last video in
+   * the playlist be the first video in the playlist.
+   *
+   * @param  {boolean} [val]
+   *         The value to set repeat to
+   *
+   * @return {boolean}
+   *         The current value of repeat
+   */
+  playlist.repeat = (val) => {
+    if (val !== undefined) {
+      playlist.repeat_ = !!val;
+    }
+    return playlist.repeat_;
+  };
+
+  /**
+   * Sorts the playlist array.
+   *
+   * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort}
+   * @fires playlistsorted
+   *
+   * @param {Function} compare
+   *        A comparator function as per the native Array method.
+   */
+  playlist.sort = (compare) => {
+
+    // Bail if the array is empty.
+    if (!list.length) {
+      return;
+    }
+
+    list.sort(compare);
 
     /**
-     * Plays the previous item in the playlist.
+     * Triggered after the playlist is sorted internally.
      *
-     * @return {Object|undefined}
-     *         Returns undefined and has no side effects if on first item.
+     * @event playlistsorted
+     * @type {Object}
      */
-    previous() {
+    player.trigger('playlistsorted');
+    return list;
```

Comment:
It almost seems weird to just return the list here as if you want to method chain you would probably want to do so with playlist methods.

Reply:
Yeah, it originally didn't return anything. I'll probably stick with that pattern.